osf_tests/factories.py

```diff
@@ -1115,3 +1115,81 @@ class Meta:
 
     primary_color = factory.Faker('hex_color')
     secondary_color = factory.Faker('hex_color')
+
+
+class RegistrationSchemaFactory(DjangoModelFactory):
+    name = factory.Faker('catch_phrase')
+    schema_version = factory.Sequence(lambda n: n)
+
+    schema = factory.Sequence(lambda n: {
+        'title': f'Test Data Schema #{n}',
+        'version': 1,
+        'config': {
+            'hasFiles': True
+        },
+        'description': f'Test Schema #{n} for internal use only',
+        'pages': [{
+            'id': 'page1',
+            'title': 'Study Information',
+            'questions': [{
+                'qid': 'q1',
+                'nav': 'Nav, what does this do?',
+                'type': 'string',
+                'format': 'textarea',
+                'title': 'Maybe a section heading',
+                'description': 'Maybe a question label.',
+                'help': 'example text for question-label',
+                'required': True
+            }, {
+                'qid': 'q2',
+                'nav': 'Hypotheses',
+                'type': 'string',
+                'format': 'textarea',
+                'title': 'Maybe a section heading',
+                'description': 'Maybe a question label.',
+                'help': 'help text',
+                'required': True
+            }]
+        }]
+    })
+
+    @classmethod
+    def _create(cls, *args, **kwargs):
+        from osf.utils.migrations import map_schema_to_schemablocks
+        schema = super()._create(*args, **kwargs)
+        map_schema_to_schemablocks(schema)
+        return schema
+
+    class Meta:
+        model = models.RegistrationSchema
+
+
+class SchemaResponseFactory(DjangoModelFactory):
+    initiator = factory.SubFactory(AuthUserFactory)
+    parent = factory.SubFactory(RegistrationFactory)
+    revision_justification = "We're talkin' about practice!"
+    submitted_timestamp = FuzzyDateTime(datetime.datetime(1970, 1, 1, tzinfo=pytz.UTC))
+    schema = factory.SubFactory(RegistrationSchemaFactory)
+
+    class Meta:
+        model = models.SchemaResponse
+
+    @classmethod
+    def _create(cls, *args, **kwargs):
+        from django.contrib.contenttypes.models import ContentType
+
+        SchemaResponse = models.SchemaResponse
+        justification = kwargs.get('revision_justification')
+        initiator = kwargs.get('initiator')
+        parent = kwargs.get('parent')
+        schema = kwargs.get('schema')
+        parent.registered_schema.add(schema)
+
+        if SchemaResponse.objects.filter(object_id=parent.id, content_type_id=ContentType.objects.get_for_model(parent)).exists():
+            previous_schema_response = SchemaResponse.objects.filter(
+                object_id=parent.id,
+                content_type_id=ContentType.objects.get_for_model(parent)
+            ).order_by('created').first()
```

Comment:
`order_by` defaults to ascending order. This should either be `order_by('-created').first()` or `order_by('created').last()`. Could also use `updated_response__isnull=True` in the filter to get the latest un-revised response (assuming we're not putting ourselves into. weird state).